operators/src/source/gdal_source/loading_info.rs

```diff
@@ -460,6 +538,47 @@ impl Iterator for NetCdfCfGdalLoadingInfoPartIterator {
 pub struct GdalLoadingInfo {
     /// partitions of dataset sorted by time
     pub info: GdalLoadingInfoTemporalSliceIterator,
+    /// To answer the query time, the `TimeInterval` of the first issued part must contain the query start and the last issued part must contain the query end.
+    /// This is the start of the first part produced to answer the query. If there is no data part that intersects the query start, this is the last time known before the query start.
+    /// IF there is a part starting exactly at the query time start. It is the start time of that part (and the query).
+    /// ELSE IF there is a known element with an end before the query time it is the end of that `TimeInterval`.
+    /// ELSE IF there is no data at all (and never will be) it might be `TimeInstance::MIN`.
+    /// ELSE it should be None and the GdalSource will replace it with the start of the query AND issue a warning about missing information.
+    pub first_part_to_answer_query_time_start: Option<TimeInstance>,
+    /// This is the end of the last part produced to answer the query.
+    /// IF there is no data part that intersects the query end, this is the first time known after the query end.
+    /// ELSE IF there is a known element with an end after the query time it is the start of that `TimeInterval`.
+    /// ELSE IF there is no data at all (and never will be) it might be `TimeInstance::MAX`.
+    /// ELSE it should be None and the GdalSource will replace it with the start of the query AND issue a warning about missing information.
+    pub last_part_to_answer_query_time_end: Option<TimeInstance>,
+}
+
+impl GdalLoadingInfo {
+    /// This method generates a new `GdalLoadingInfo`.
```

Comment:
maybe start by explaining that the temporal slice iterator may not start/end with the query. in this case the first temporal slice starts after the query start and there is a gap that has to be filled with a nodata tile. This nodata tile needs a temporal validity, thus we need to know how long before the start of the first loading info there is no data. This information is passed as known_time_before/after

but also think about renaming this fields to something like:
- start_time_of_output_stream
- result_temporal_validity (and combining both time instances into a timeinterval)

Reply:
[5a4c3e1](https://github.com/geo-engine/geoengine/pull/968/commits/5a4c3e1d58f61a93f49a9e7a07c1e45bafe5befe)